sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/init.sql

```diff
@@ -189,13 +190,57 @@ metrics_ping AS (
     )[SAFE_OFFSET(0)] AS adjust_creative,
     ARRAY_AGG(metrics.string.metrics_install_source IGNORE NULLS ORDER BY submission_timestamp ASC)[
       SAFE_OFFSET(0)
-    ] AS install_source
+    ] AS install_source,
+    ARRAY_AGG(
+      metrics.string.metrics_adjust_ad_group IGNORE NULLS
+      ORDER BY
+        submission_timestamp DESC
+    )[SAFE_OFFSET(0)] AS latest_adjust_ad_group,
+    ARRAY_AGG(
+      metrics.string.metrics_adjust_creative IGNORE NULLS
+      ORDER BY
+        submission_timestamp DESC
+    )[SAFE_OFFSET(0)] AS latest_adjust_creative,
+    ARRAY_AGG(
+      metrics.string.metrics_adjust_network IGNORE NULLS
+      ORDER BY
+        submission_timestamp DESC
+    )[SAFE_OFFSET(0)] AS latest_adjust_network,
+    ARRAY_AGG(
+      metrics.string.metrics_adjust_campaign IGNORE NULLS
+      ORDER BY
+        submission_timestamp DESC
+    )[SAFE_OFFSET(0)] AS latest_adjust_campaign,
   FROM
     org_mozilla_firefox.metrics AS org_mozilla_firefox_metrics
   WHERE
     DATE(submission_timestamp) >= '2019-01-01'
   GROUP BY
     client_id
+),
+-- Find most recent client details from the baseline ping.
+baseline_ping AS (
+  SELECT
+    client_info.client_id,
+    DATE(MAX(submission_timestamp)) AS last_seen_date,
+    ARRAY_AGG(normalized_country_code IGNORE NULLS ORDER BY submission_timestamp DESC)[
```

Comment:
`HAVING MAX` - https://cloud.google.com/bigquery/docs/reference/standard-sql/aggregate-function-calls#having_max is now available. Which seems to be functionally equivalent here. I used this for consistency

Reply:
I'm into the `HAVING MAX`, since it's shorter (and _may_ be more efficient, depends on if BQ is smart enough to realize it doesn't have to materialize the interim array in ARRAY_AGG).